### ISSUE TITLE
NPM package incorrectly contains a nested @types/react package

### ISSUE THREAD

**Issue by UselessPickles** [react-monaco-editor/react-monaco-editor#256]
**Describe the bug**
The NPM package incorrectly contains a nested @types/react package. This causes compiler errors in react+typescript projects because typescript finds two .d.ts files for `react` and attempts to de-declare all the types.

I also noticed that `@types/react` is incorrectly identified as a "dependency" in package.json. It should instead be moved to "devDependencies". I'm not sure if this is the entirety of the fix to this problem, or if there is a manual process of creating the package for NPM that causes `@types/react` to be included with the package.

**To Reproduce**
npm install `react-monaco-editor` and observe:
![Screen Shot 2019-10-31 at 11 07 46 AM](https://user-images.githubusercontent.com/22800095/67961204-c39b9200-fbd1-11e9-8298-0f9ebb528662.png)

**Expected behavior**
The `react-monaco-editor` distribution should not contain a copy of `@types/react`

**Comment by UselessPickles** [comment]
This is a critical blocker for me. The only workaround I have found is to enable the "skipLibChecks" option for typescript, but I really don't want to disable typechecking for all libraries.

**Comment by UselessPickles** [comment]
Maybe `@types/react` needs to be a devDependency and a peerDependecy? For building your package, you can depend on a specific version of react types, but you should not force others to use the same exact type.

**Comment by UselessPickles** [comment]
Actually, that doesn't make sense either. It's not really a peerDependency either, because this package does not require others to use typescript. I think how I have it as a devDependecy is correct.

**Comment by UselessPickles** [comment]
Its up to the consuming project of this library to choose to use typescript, and it's then that project's responsibility to include correct typings  for things like react.

**Comment by UselessPickles** [comment]
Since your project is compiled using react "^16.8.3", you have to have a devDependency of the correct corresponding version of `@types/react`. You can't change your dependency on `@types/react` to be "^15" because then your project would have incorrect typings

**Comment by UselessPickles** [comment]
Why was this closed?

**Comment by UselessPickles** [comment]
I see you changed the dependency, but you still have a problem. This package will not work in projects that use typescript and react 15.x:

```
  "peerDependencies": {
    "monaco-editor": "*",
    "react": "^15.x || ^16.x" // <--- allows for react 15.x
  },
  "dependencies": {
    "@types/react": "^16.0.0", // <-- not compatible with react 15.x
    "prop-types": "^15.0.0"
  }
```

You really need to move `@types/react` to devDependencies to make this work correctly. Or maybe you need to drop support for react 15.x?

**Comment by UselessPickles** [comment]
I think the most obvious indicator that "dependencies" is the wrong place is that typings are not required for projects that do not use typescript, and typescript is not required to use this package, so you should not force all consumers of this package to depend on typescript typings.

**Comment by domoritz** [comment]
> Its up to the consuming project of this library to choose to use typescript, and it's then that project's responsibility to include correct typings for things like react.

I'm not sure I agree here. I think react typings are an optional dependency. If someone uses Typescript, they should get the right version of the typings. 

> You really need to move @types/react to devDependencies to make this work correctly. Or maybe you need to drop support for react 15.x?

Let me add 15.x. I released this in 0.32.1. Is that better?
